src/core/execute.c

```diff
@@ -1879,19 +1879,36 @@ void exec_status_dump(const ExecStatus *s, FILE *f, const char *prefix) {
                         "%sStart Timestamp: %s\n",
                         prefix, FORMAT_TIMESTAMP(s->start_timestamp.realtime));
 
-        if (dual_timestamp_is_set(&s->handoff_timestamp))
+        if (dual_timestamp_is_set(&s->handoff_timestamp) && dual_timestamp_is_set(&s->start_timestamp) && s->handoff_timestamp.monotonic > s->start_timestamp.monotonic)
+                fprintf(f,
+                        "%sHandoff Timestamp: %s (%s since start)\n",
+                        prefix, FORMAT_TIMESTAMP(s->handoff_timestamp.realtime), FORMAT_TIMESPAN(usec_sub_unsigned(s->handoff_timestamp.monotonic, s->start_timestamp.monotonic), 1));
```

Comment:
Linebreaks please.

Hmm, for most services, all the timestamps are going to be the same, since it'd be strange for start to take more than a second. So actually the timing information will only be available via the diff (nnn ms since start), which is rather awkward. So I think it'd be better to format all timestamps here as numerical value (seconds.microseconds since epoch). In practice this will be much more useful than the "pretty" timestamp. This is a debug interface after all.

Reply:
dunno, i think showing an "anchor" in calendar time definitely makes sense, so you know the range we are talking off.

Showing the time difference like this does here is really useful as it shows that our executor binary sometimes is super slow. At least here i found a couple of cases where it took 750ms between start + handoff, which is quite a lot I'd say. For completely benign Type=oneshot services. Seems like something to debug eventually. Anyway long story short: I think the time difference this now shows is very valuable.

Reply:
I think you misunderstood my comment. Showing the diff is fine. I have an issue with how the main timestamp is shown.

I see something like this:
```
        → ExecStart:
                Command Line: /usr/sbin/logrotate /etc/logrotate.conf
                        PID: 4018609
                        Start Timestamp: Thu 2024-04-25 00:00:08 CEST
                        Handoff Timestamp: Thu 2024-04-25 00:00:08 CEST (33.654ms since start)
                        Exit Timestamp: Thu 2024-04-25 00:00:08 CEST (9.558ms since handoff)
                        Exit Code: exited
                        Exit Status: 0
```
With one-second granularity, the timestamps will often be the same string repeated three times.

The purpose of those timestamps is mainly timing. Thus, I think a time format that doesn't show the week day and time zone, but instead shows the precise time, would be more useful. Something like our short-monotonic or short-precise time formats.

Reply:
I am not sure i follow. I am pretty sure we should show calendar time at least once? but suppressing the calendar time if mostly the same as the others is really messy.

"short-monotonic" and "short-precise" are log message formatting styles. it's not a concept in our codebase for timestamps. I am not sure it should be, and I am not to keen into turning this into that.

Also, the "since start" + "since handoff" lines already make the monotonic clock very visible, in a more useful way, i.e. just showing the relative diff, not the absolute values. in combination with the calendar time we show this should cover all bases, for long and for short running stuff.

this is debug output, i am not sure this really needs to be tweaked that much.

Reply:
I changed the code now to show only the diff on the monotonic clock if we have it, and calendar otherwise. I don't think it's really that great, given that we just suppress valuable info in the the dump now, but if that gets this merged...